_posts/2020-06-21-one-stop-guide-to-lombok.md

```diff
@@ -0,0 +1,337 @@
+    }
+}
+```
+
+## @Data annotation
+This annotation is used the most frequently amongst all other Lombok annotations. It combines the functionality of the next Lombok annotations: `@ToString`, `@EqualsAndHashCode`, `@Getter`, `@Setter`,
+`@RequiredArgsConstructor`
+
+# How to add the support of Lombok in your project
+
+## Dependencies to add for Maven
+
+To set up lombok with Maven, you have to specify that the lombok dependency is required to compile your source code, but does not need to be present when running/testing/jarring/otherwise deploying your code. Generally this is called a 'provided' dependency.  
+Lombok is available in maven central, so you can easily add it in `dependencies` section of your `pom.xml` file.
+
+```java
+<dependencies>
+  <dependency>
+    <groupId>org.projectlombok</groupId>
+    <artifactId>lombok</artifactId>
+    <version>1.18.12</version>
+    <scope>provided</scope>
+  </dependency>
+</dependencies>
+```
+
+*N.B.* ~when adding lombok annotations make sure that you have imported them from correct package. This is the correct import for annotation `@Getter`  
+`import lombok.Getter`~
+
+If you use Jdk9+ you should add the following to the maven compiler plugin configuration section:  
+```java
+<annotationProcessorPaths>
+  <path>
+    <groupId>org.projectlombok</groupId>
+    <artifactId>lombok</artifactId>
+    <version>1.18.12</version>
+  </path>
+</annotationProcessorPaths>
+```
+
+## Dependencies to add for Gradle
+
+As with setting up lombok for Maven you have to specify lombok dependency in your `buiild.gradle`
+
+In case you do not want to use Lombok Gradle plugin gradle has the built-in `compileOnly` scope, which can be used to tell Gradle to add lombok only during compilation.
+
+```java
+dependencies {
+  compileOnly 'org.projectlombok:lombok:1.18.12'
+  annotationProcessor 'org.projectlombok:lombok:1.18.12'
+}
+```
+
+If you want to use lombok gradle plugin you can read more about it [https://plugins.gradle.org/plugin/io.freefair.lombok](here)
+
+## Settings for IntelliJ Idea to activate Lombok preprocessing
+
+To use Lobmok features you have to activate preprocessing of Lombok annotations. For Intellij you have to go to  
+~`Settings > Build, Execution, Deployment > Compiler > Annotation Processors`~ and activate `Enable annotation processing`
+
+The next popular features is `@EqualsAndHashCode`. With this feature usually `@EqualsAndHashCode.Exclude` used to exclude some fields from using in `equals()` and `hashCode()` methods.
+
+# Caveats when using some features of Lombok
+
+Using Lombok annotations is easy. But sometimes there are some caveats. They often can be met when you have extended from some superclass.
+In those cases you can use some additional parameter like `callSuper=true`. This parameter can be used in such features like `@ToString, @EqualsAndHashCode etc.`. Using or not the parameter `callSuper` can be not such harmful in `@ToString` as it can be in `@EqualsAndHashCode`.
```

Comment:
Why is inheritance problematic? Why is it more problematic in EqualsAndHashCode than in ToString? What does it do at all? Add a code example with `callSuper`, that would make it clearer.

Reply:
Please do a little more research as to what others think why Lombok can be problematic and perhaps link to those articles. These are the things that make this article stand out. Otherwise, this is just a copy of the Lombok docs.